lib/modules/dosomething/dosomething_reportback/includes/ReportbackItem.php

```diff
@@ -115,8 +115,16 @@ private function build($data) {
         'verb' => $data->verb,
       ],
     ];
+
+    $northstar_user = dosomething_northstar_get_northstar_user($data->uid);
+    $northstar_user = json_decode($northstar_user->data, true);
+
```

Comment:
Not vital at all (or maybe just a future update to consider), but might be cleaner if you could save the nested array to `$northstar_user` and then cast it to an object:

```php
$northstar_user = json_decode($northstar_user->data, true);
$northstar_user = (object) $northstar_user['data'][0];
```

So then you can get each item similar to how code above retrieves it for consistency:

```php
$this->user = [
  'id' => $data->uid,
  'first_name' => $northstar_user->first_name,
  'last_name' => $northstar_user->last_name,
  'photo' => $northstar_user->photo,
  'country' => $northstar_user->country,
];
```

Something like that... :)